docs/content/docs/dev/table/types.md

```diff
@@ -1487,6 +1481,88 @@ Not supported.
 {{< /tab >}}
 {{< /tabs >}}
 
+Casting
+-------
+
+Flink SQL can perform casting between a defined pair `from` type and `target` type pair. While some
+casting operations can always succeed regardless of the input value, others can fail at runtime
+(i.e. where there is no way to create a value for the target type). For example, it is always
+possible to convert `INT` to `STRING`, but you cannot always convert a `STRING` to `INT`.
+
+During the planning stage, Flink SQL rejects queries for invalid cast input and target type with
+a `ValidationException`, e.g. when trying to cast a `TIMESTAMP` to an `INTERVAL`.
+While for valid type pairs that can fail at runtime Flink SQL accepts them, but requires the user to correctly handle failures.
```

Comment:
```suggestion
Valid type pairs that can fail at runtime will be accepted by Flink SQL, but requires the user to correctly handle failures.
```

Reply:
Repeating every time _Table API and SQL_ is a bit exagerated imo, i used _query validator_ instead.